### PR TITLE
fix #1959, #2002, VectorGraphics minor update, expression mutableMatrix

### LINKED ISSUE #1959: infinite loop in Matrix//Number
```
i1 : matrix {{1,2},{3,4}}

o1 = | 1 2 |
     | 3 4 |

              2        2
o1 : Matrix ZZ  <--- ZZ

i2 : oo//2
stdio:2:3:(3): error: recursion limit of 300 exceeded

i3 : code(symbol //,Matrix,Number)

o3 = -- code for method: Matrix // Number
     ../../../M2/M2/Macaulay2/m2/matrix2.m2:360:27-360:50: --source code:
     Matrix // Number := (f,r) -> f // promote(r,ring f)
```

### LINKED ISSUE #2002: RingMap not promoting matrices
I see no reason why `o4` shouldn't work. I think one could easily change `RingMap Matrix` to behave more like `RingMap RingElement` and allow promotion.

```
i1 : R=QQ[x]; f=map(R,R)

o2 = map (R, R, {x})

o2 : RingMap R <--- R

i3 : f 1

o3 = 1

o3 : R

i4 : f matrix{{1}}
stdio:3:1:(3): error: expected source of ring map to be the same as ring of matrix
```

### PR DESCRIPTION
**Pull request by pzinn**
- fix #1959. I understand that eventually one may want to rethink how the two types `Number` and `RingElement` interact, but in the meantime we need to fix this.
- fix #2002. I did not rewrite the part of mutable matrices due to #2192.
- value expression MutableMatrix now correctly produces a mutable matrix.
- minor VectorGraphics update

### CODE REVIEW

**Comment by pzinn** [comment]
I'm throwing in an extra bug fix:
```
R=QQ[x_(1,{1}),x_(1,{2})]
describe R
stdio:2:1:(3): error: no method for binary operator + applied to objects:
--            {1} (of class List)
--      +     1 (of class ZZ)
```

**Comment by DanGrayson** [comment]
That's strange -- 5 files in the emacs submodule are updated by your pull request, but there are no actual changes to the files.  Oh, well.

File: M2/Macaulay2/m2/expressions.m2 [hunk 1]
```diff
@@ -533,21 +533,21 @@ toString'(Function, SparseMonomialVectorExpression) := (fmt,v) -> toString (
 MatrixExpression = new HeaderType of Expression
 MatrixExpression.synonym = "matrix expression"
 matrixOpts := x -> ( -- helper function
-    opts := hashTable{CompactMatrix=>compactMatrixForm,BlockMatrix=>null,Degrees=>null};
+    opts := hashTable{CompactMatrix=>compactMatrixForm,BlockMatrix=>null,Degrees=>null,mutable=>false};
```
Comment (DanGrayson) [inline M2/Macaulay2/m2/expressions.m2:536]: How about using `MutableMatrix` instead?  That way we can adhere to the convention that optional arguments have capitalized names.

Comment (pzinn) [reply M2/Macaulay2/m2/expressions.m2:536]: oops, good point

**Comment by d-torrance** [comment]
> That's strange -- 5 files in the emacs submodule are updated by your pull request, but there are no actual changes to the files. Oh, well.

I think that must be a bug with GitHub, as there are definitely changes:

```
profzoom@peg-amy:~/src/macaulay2/M2-emacs$ git diff --stat 2578f39 bffc2c7
 D-mode.el         |   4 ++--
 M2-emacs-help.txt |   4 ++--
 M2-emacs.m2       |   2 +-
 M2-symbols.el.gz  | Bin 10850 -> 11094 bytes
 M2.el             |   2 +-
 5 files changed, 6 insertions(+), 6 deletions(-)
```

**Comment by pzinn** [comment]
I certainly didn't mean to change any files in M2-emacs. I'll look into it.

**Pull request merged**